source/intune/IntuneClient.cs

```diff
@@ -1316,5 +1365,136 @@ public async Task SyncDevices(List<IntuneDevice> devices, LiteDBHandler database
             }
             Logger.Info("Successfully sent request for device sync notification");
         }
+
+        public async Task<string[]> CreateDCv1DiagnosticLogPolicy(string deviceId, string url, List<string> registryKeys, List<string> events, List<string> commands, List<string> folderFiles, string outputFileFormat)
+        {
+            Logger.Info($"Creating custom config policy for device: {deviceId}");
+            string endpoint = "https://graph.microsoft.com/beta/deviceManagement/deviceConfigurations";
+
+            var xmlDoc = new System.Xml.XmlDocument();
+            var root = xmlDoc.CreateElement("Collection");
+
+            var idElement = xmlDoc.CreateElement("ID");
+            var requestId = Guid.NewGuid().ToString();
+            idElement.InnerText = requestId;
+            root.AppendChild(idElement);
+
+            var urlElement = xmlDoc.CreateElement("Url");

```

Comment:
```suggestion
            var urlElement = xmlDoc.CreateElement("Url");
            var urlElement = xmlDoc.CreateElement("SasUrl");
```
@Mayyhem I knew I should've tested end-to-end... sorry about that! let me know if it doesn't work, and I'll test in my env too. sounds like there's a bug in the CSP provider as well, since the invalid payload reported a success for you